appinventor/blocklyeditor/src/blocklyeditor.js

```diff
@@ -156,6 +157,7 @@ Blockly.BlocklyEditor.render = function() {
   var timeDiff = stop - start;
   Blockly.Instrument.stats.totalTime = timeDiff;
   Blockly.Instrument.displayStats("Blockly.BlocklyEditor.render");
+  Blockly.WarningHandler.checkAllBlocksForWarningsAndErrors();
```

Comment:
Hi Jeff,

This line was added to update the warning when users add a component for the block that its component is missing. In other words, this is how it would work. 

- When users paste a block from Backpack of which component does not exist, the red flag will be raised ' the component does not exist'. 
- Once the users add the missing component, then the red flag will be disappear (the line 160 checks errors and warnings and remove unnecessary flags)

Without the line, still the warning will be updated when users click the block or anywhere in workspace. This may not be as intuitive as updating the flags right away, but it does the job.

Let me know if it is okay without the line 160, or there is a better way to do it. 

http://grace-ai2.appspot.com/
This is the Backpack instance and I just commented out the line160. 

Best,
Grace 

@jisqyv @ram8647